src/main/java/org/elasticsearch/index/query/MatchAllQueryParser.java

```diff
@@ -44,32 +42,28 @@ public MatchAllQueryParser() {
         return new String[]{NAME, Strings.toCamelCase(NAME)};
     }
 
-    @Override
-    public Query parse(QueryParseContext parseContext) throws IOException, QueryParsingException {
+    public MatchAllQueryBuilder fromXContent(QueryParseContext parseContext) throws IOException {
+        MatchAllQueryBuilder queryBuilder = new MatchAllQueryBuilder();
         XContentParser parser = parseContext.parser();
 
-        float boost = 1.0f;
         String currentFieldName = null;
-
         XContentParser.Token token;
         while (((token = parser.nextToken()) != XContentParser.Token.END_OBJECT && token != XContentParser.Token.END_ARRAY)) {
             if (token == XContentParser.Token.FIELD_NAME) {
                 currentFieldName = parser.currentName();
             } else if (token.isValue()) {
                 if ("boost".equals(currentFieldName)) {
-                    boost = parser.floatValue();
+                    queryBuilder.boost = parser.floatValue();
                 } else {
                     throw new QueryParsingException(parseContext.index(), "[match_all] query does not support [" + currentFieldName + "]");
                 }
             }
         }
+        return queryBuilder;
+    }
 
-        if (boost == 1.0f) {
-            return Queries.newMatchAllQuery();
-        }
-
-        MatchAllDocsQuery query = new MatchAllDocsQuery();
-        query.setBoost(boost);
-        return query;
+    @Override
+    public Query parse(QueryParseContext parseContext) throws IOException, QueryParsingException {
```

Comment:
Having BaseQueryParser was exactly yo make sure that you don't have to repeat this method all over. see my comment above